tools/codegen/pkg/schemapatch/schemapatch.go

```diff
@@ -105,3 +120,19 @@ func loadGroupRuntime(paths []string) (*genall.Runtime, error) {
 	generators := &genall.Generators{}
 	return generators.ForRoots(paths...)
 }
+
+// outputToBuffer is a WriteCloser that writes to a buffer.
+// This is used as the output for the controller-gen generator integration.
+type outputToBuffer struct {
+	*bytes.Buffer
+}
+
+// Open implements the Open method of the io.WriteCloser interface.
+func (o *outputToBuffer) Open(_ *loader.Package, _ string) (io.WriteCloser, error) {
+	return o, nil
+}
+
+// Close implements the Close method of the io.WriteCloser interface.
+func (o outputToBuffer) Close() error {
```

Comment:
Is there a particular reason to have a value receiver here?

Reply:
Since it doesn't actually do anything, it can be either value or pointer, I have no strong preference but  this works for both values and pointers, so, is better? 🤷

Reply:
With value receiver, it will copy the buffer with no reason, as far as I understand this.